Mangosta/ViewControllers/LoginViewController.swift

```diff
@@ -22,11 +27,33 @@ class LoginViewController: UIViewController {
 			let auth = AuthenticationModel(jidString: self.jidField.text!, password: self.passwordField.text!)
 			auth.save()
 		}
+        
+
+        self.configureAndStartXMPP()
 
-		self.loginDelegate?.didPressLogInButton()
-		self.dismissViewControllerAnimated(true, completion: nil)
 	}
     
+    func configureAndStartXMPP() {
+        
+        let authModel = AuthenticationModel.load()!
+        
+        self.xmppController = XMPPController(hostName: authModel.serverName!,
+                                             userJID: authModel.jid,
+                                             password: authModel.password)
+        
+        let appDelegate = UIApplication.sharedApplication().delegate as! AppDelegate
+        appDelegate.xmppController = self.xmppController
```

Comment:
Be aware of retain cycles with the `.xmppController` variable in `AppDelegate`. Make sure it's a `weak` reference.